src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs

```diff
@@ -138,6 +146,20 @@ private static void EnsureInitialize(WebHostSettings settings)
             }
         }
 
+        private static void ReinitializeAppSettings()
+        {
+            // only in azure environment
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("WEBSITE_SITE_NAME")))
```

Comment:
My bad, forgot this wasn't Kudu! Might be worth adding a helper if there are other places in the code where we check for Azure.